project/CONTRIBUTING.md

```diff
@@ -299,3 +300,42 @@ This is a consensus made on #4242 in addition to #1337.
 We agreed that blank color is boring and so deja vu. Pink is the new way to do.
 ```
 (Obviously, this commit is fake. :wink:)
+
+## Drafting a new release
+
+### Minor releases and patch releases
+
+Releasing software is the last step to getting your bugfixes or new features to your user base,
+and should be done regularly, so that users are not tempted to use development branches.
+To know what you are going to release on branch 42.x, given that the last release on this branch is 42.3.1,
+go to `https://github.com/sonata-project/SonataCoreBundle/compare/42.3.1...42.x`.
+You should see a list of commits, some of which should contain links to pull requests.
+
+#### Determining the next release number
+
+If any of those pull requests is labeled `minor`, then the next release should be a minor release (42.4.0).
+In the opposite case, and if any of those pull requests is labeled `patch`,
+then the next release should be a patch release (42.3.2).
+If none of the conditions above are fulfilled, then all the commits should be labeled either `docs` or `pedantic`,
+and you should not make a release.
+
+#### Adding the release to the UPGRADE file
+
+If there are any deprecations, then the release should be minor and the UPGRADE-42.x file should be changed,
+moving the instructions explaining how to bypass the deprecation messages,
+that should hopefully be there, to a new section named `UPGRADE FROM 42.3.1 to 42.4.0`.
+
+#### Compiling the changelog
+
+Each non-pedantic PR should contain a `CHANGELOG` section, that you should compile together in the `CHANGELOG.md` file,
+under a title in the following format : `[42.3.2](https://github.com/sonata-project/SonataNewsBundle/compare/42.3.1...42.3.2) - YYYY-MM-DD`.
```

Comment:
compiling the changelog is thus a manual thing, right? maybe say "... that you need to copy manually into the `CHANGELOG.md` file. The title is in the following format: ..."